otx/core/data/segmentation_dataset_adapter.py

```diff
@@ -0,0 +1,79 @@
+"""Segmentation Dataset Adapter."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# pylint: disable=invalid-name, too-many-locals, no-member, too-many-nested-blocks
+from typing import Any, Tuple
+
+from datumaro.components.annotation import Mask as DatumaroMask
+from datumaro.plugins.transforms import MasksToPolygons
+
+from otx.api.entities.annotation import (
+    Annotation,
+    AnnotationSceneEntity,
+    AnnotationSceneKind,
+    NullAnnotationSceneEntity,
+)
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.image import Image
+from otx.api.entities.label_schema import LabelSchemaEntity
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.polygon import Point, Polygon
+from otx.core.data.base_dataset_adapter import BaseDatasetAdapter
+
+
+class SegmentationDatasetAdapter(BaseDatasetAdapter):
+    """Segmentation adapter inherited by BaseDatasetAdapter.
+
+    It converts DatumaroDataset --> DatasetEntity for semantic segmentation task
+    """
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Segmentation."""
+        # Prepare label information
+        label_information = self._prepare_label_information(datumaro_dataset)
+        label_entities = label_information["label_entities"]
+
+        # Label schema
+        label_schema = self._generate_default_label_schema(label_entities)
+
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for _, datumaro_items in subset_data.subsets().items():
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
+                    shapes = []
+                    for ann in datumaro_item.annotations:
+                        if isinstance(ann, DatumaroMask):
```

Comment:
Same here, can we eliminate isinstance? I'd consider appearing of non-segmentation annotation objects as a bug (I guess we use this converter in OTX CLI scenario only).